packages/vite-plugin-cloudflare/src/shared.ts

```diff
@@ -1,5 +1,5 @@
-import { MODULE_TYPES } from "./constants";
+import { ADDITIONAL_MODULE_TYPES } from "./constants";
 
 export const UNKNOWN_HOST = "http://localhost";
 export const INIT_PATH = "/__vite_plugin_cloudflare_init__";
-export const MODULE_PATTERN = `__CLOUDFLARE_MODULE__(${MODULE_TYPES.join("|")})__(.*?)__`;
+export const ADDITIONAL_MODULE_PATTERN = `__CLOUDFLARE_MODULE__(${ADDITIONAL_MODULE_TYPES.join("|")})__(.*?)__`;
```

Comment:
This pattern is only ever used to create Regexes.
I am not 100% sure whether V8 is able to cache such things but it feels wasteful to have to create a new regex for every usage, especially inside hooks like `renderChunk`.
How about we create the Regex once here and reuse that instead of the string?

Reply:
I did originally have this similar to how you suggest but you then need to create a separate regex here for the global and non-global variants and remember to set `lastIndex = 0` when using the global one. Happy to change it back though.

Reply:
Just thinking about this. Now that I've made `renderChunk` async I think that reusing the same regex instance could lead to unexpected behaviour because of shared `lastIndex` state. Using a new regex instance inside `renderChunk` is a lot easier to reason about. I could still create a single regex instance to use for the non global usage elsewhere if you prefer.

Reply:
Ah yes, of course. Definitely don't want to share the regex state for this. OK.

Reply:
Made some changes to reuse the regex for the non-global cases.

[18a609d](https://github.com/cloudflare/workers-sdk/pull/8387/commits/18a609d88e0a682659b3976221250766ca14eeaf)